bftengine/src/bcstatetransfer/InMemoryDataStore.cpp

```diff
@@ -237,9 +235,10 @@ void InMemoryDataStore::setResPage(uint32_t inPageId,
                                    const char* inPage) {
   auto lock = std::unique_lock(reservedPagesLock_);
   LOG_DEBUG(logger(), "pageId: " << inPageId << " checkpoint: " << inCheckpoint);
-  // create key, and make sure that we don't already have this element
+  // create key
+  // pages may already contain key when restarted replica asks and gets
+  // last stored checkpoint from network which is also stopped at same checkpoint
   ResPageKey key = {inPageId, inCheckpoint};
-  ConcordAssert(pages.count(key) == 0);
```

Comment:
same idea like previous comment. lets try to keep it, and zero the res page from outside, or (less preferable) tell to jump assert using a flag.